thrift.js

```diff
@@ -46,6 +49,10 @@ var ThriftMap = require('./map').ThriftMap;
 var ThriftConst = require('./const').ThriftConst;
 var ThriftTypedef = require('./typedef').ThriftTypedef;
 
+var casThriftCache = {};
```

Comment:
It pains me to say, this is not the right problem for a content address store. It is valid for multiple documents to have identical content, and hashing every included file is overkill. Not even necessary to canonicalize the path of the file for this case, quite sufficient to fully qualify the include path and use that as the cache key.